pkg/quota_management/organisation.go

```diff
@@ -19,16 +34,52 @@ func (org Organisation) HasUsersRegistered() bool {
 	return len(org.RegisteredUsers) > 0
 }
 
-func (org Organisation) IsInstanceCountWithinLimit(count int) bool {
-	return count <= org.GetMaxAllowedInstances()
+func (org Organisation) IsInstanceCountWithinLimit(instanceTypeID string, billingModelID string, count int) bool {
+	return count <= org.GetMaxAllowedInstances(instanceTypeID, billingModelID)
+}
+
+func (org Organisation) GetMaxAllowedInstances(instanceTypeID string, billingModelID string) int {
+	bm, ok := org.getBillingModel(instanceTypeID, billingModelID)
+
+	if !ok {
+		return 0
+	}
+
+	if bm.Allowed <= 0 {
+		if org.MaxAllowedInstances <= 0 {
+			return MaxAllowedInstances
+		}
+		return org.MaxAllowedInstances
+	}
+
+	return bm.Allowed
+}
+
+func (org Organisation) GetGrantedQuota() QuotaList {
+	if len(org.GrantedQuota) == 0 {
+		return defaultQuotaList
+	}
+	return org.GrantedQuota
 }
 
-func (org Organisation) GetMaxAllowedInstances() int {
-	if org.MaxAllowedInstances <= 0 {
-		return MaxAllowedInstances
+func (org Organisation) getBillingModel(instanceTypeId string, billingModelID string) (BillingModel, bool) {
```

Comment:
depending on some answers on other comments this might be better reworded as getBillingModelQuotaEntitlement or getBillingModelGrantedQuota. Same applies to the `BillingModel` itself

Reply:
I am not sure making the rewording is redundant. Due to the term "billing model" is overloaded through the code, and the source of truth of the accepted values belong to different packages I think it might be good to be more explicit.

From the comment https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1368#discussion_r1023146896 I understand this would be kafka billing model.

Unless we don't want to tie the name to it. But then at least we should be very clear that on the current implementation that's the expectation